web/lib/sankeyDataProcessor.ts

```diff
@@ -0,0 +1,308 @@
+/**
+ * Data processing utilities for the Sankey diagram
+ */
+
+// Original mining event interface
+export interface MiningEvent {
+  poolName: string;
+  merkleBranches: string[];
+}
+
+// Real event data format from the global data stream
+export interface StratumV1Event {
+  type: string;
+  id: string;
+  timestamp: string;
+  data: {
+    _id: string;
+    timestamp: string;
+    pool_name: string;
+    height: number;
+    job_id: string;
+    prev_hash: string;
+    coinbase1: string;
+    coinbase2: string;
+    merkle_branches: string[];
+    version: string;
+    nbits: string;
+    ntime: string;
+    clean_jobs: boolean;
+    extranonce1: string;
+    extranonce2_length: number;
+  };
```

Comment:
I have this already mapped out in `./types`, so you can probably just add an:
```
import { StratumV1Data } from "./types";
```
and then:
```suggestion
  data: StratumV1Data;
```